src/libraries/System.Private.CoreLib/src/System/Globalization/Ordinal.Utf8.cs

```diff
@@ -0,0 +1,332 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Intrinsics;
+using System.Text;
+using System.Text.Unicode;
+
+namespace System.Globalization
+{
+    internal static partial class Ordinal
+    {
+        internal static int CompareStringIgnoreCaseUtf8(ref byte strA, int lengthA, ref byte strB, int lengthB)
+        {
+            int length = Math.Min(lengthA, lengthB);
```

Comment:
Nit: Consider putting a comment here that two UTF-8 inputs of different length might compare as equal under an OrdinalIgnoreCase comparer. This is distinct from UTF-16, where the inputs being different length will mean that they can never compare as equal under an OrdinalIgnoreCase comparer.

The logic of this method looks correct, but IMO a comment such as this might help prevent people from making incorrect optimizations in the future.

Reply:
Added.